master/internal/api_logretention.go

```diff
@@ -0,0 +1,32 @@
+package internal
+
+import (
+	"context"
+
+	"github.com/determined-ai/determined/master/internal/cluster"
+	"github.com/determined-ai/determined/master/internal/grpcutil"
+	"github.com/determined-ai/determined/master/internal/logretention"
+	"github.com/determined-ai/determined/proto/pkg/apiv1"
+)
+
+func (a *apiServer) CleanupLogs(
+	ctx context.Context, req *apiv1.CleanupLogsRequest,
+) (*apiv1.CleanupLogsResponse, error) {
+	u, _, err := grpcutil.GetUser(ctx)
+	if err != nil {
+		return nil, err
+	}
+	// Check if the user has permission to update the server config, then they should be able to
+	// change the schedule and cleanup logs immediately.
+	// TODO: Update to IsClusterAdmin eventually.
+	permErr, err := cluster.AuthZProvider.Get().CanUpdateMasterConfig(ctx, u)
+	if err != nil {
+		return nil, err
+	} else if permErr != nil {
+		return nil, permErr
+	}
+
+	resp := &apiv1.CleanupLogsResponse{}
+	resp.RemovedCount, err = logretention.DeleteExpiredTaskLogs(a.m.taskSpec.LogRetentionDays)
```

Comment:
maybe we chatted about this, but maybe this should be async.

Reply:
If it took a long time then perhaps we could get a socket timeout before it completes, and if using the request context we would also cancel the delete.